### ISSUE TITLE
Wrong Simpo loss in the code

### ISSUE THREAD

**Issue by ZetangForward** [fh2c1/Anonymize-Anyone#2]
I notice in the code:

```python
# Compute losses.
model_losses = F.mse_loss(model_pred.float(), target.float(), reduction="none")
model_losses = model_losses.mean(dim=list(range(1, len(model_losses.shape))))
model_losses_w, model_losses_l = model_losses.chunk(2)

# For logging
raw_model_loss = 0.5 * (model_losses_w.mean() + model_losses_l.mean())
model_diff = model_losses_w - model_losses_l  # These are both LBS (as is t)

# Final loss by SimPO 
pi_logratios = model_losses_l - model_losses_w
logits = pi_logratios - args.gamma_beta_ratio

if args.loss_type == "sigmoid":
    losses = -F.logsigmoid(args.beta * logits)
    loss = losses.mean()
elif args.loss_type == "hinge":
    losses = torch.relu(1 - args.beta * logits)
    loss = losses.mean()                
else:
    raise ValueError(f"Unknown loss type {args.loss_type}")

implicit_acc = (pi_logratios > 0).float().mean()
implicit_acc += 0.5 * (logits == 0).sum().float() / logits.size(0)
```

`pi_logratios = model_losses_l - model_losses_w`—is this code correct? Shouldn't it be `pi_logratios = model_losses_w - model_losses_l` instead?

**Comment by fh2c1** [comment]
The calculation `pi_logratios = model_losses_l - model_losses_w` is theoretically correct. Here's why:

1. Relationship between Losses and Rewards:
- Diffusion models use MSE loss values
- Loss values are better when lower (opposite of rewards)
- Converted to rewards: reward = -loss

2. Preference Modeling:
- model_losses_w: Loss for preferred (winning) samples (should be lower)
- model_losses_l: Loss for non-preferred (losing) samples (should be higher)

3. Theoretical Verification:
- What we want: reward_w > reward_l
- Expressed in losses: -model_losses_w > -model_losses_l
- This equals model_losses_w < model_losses_l
- Therefore, model_losses_l - model_losses_w > 0 should hold

This formula becomes positive when the loss of preferred samples is lower than non-preferred samples, confirming the learning direction is correct. The code's `implicit_acc = (pi_logratios > 0).float().mean()` part also supports this logic.
